android/src/main/java/com/facebook/react/ReactFragmentActivity.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ * <p>
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.app.FragmentActivity;
+import android.view.KeyEvent;
+
+import com.facebook.react.modules.core.DefaultHardwareBackBtnHandler;
+import com.facebook.react.modules.core.PermissionAwareActivity;
+import com.facebook.react.modules.core.PermissionListener;
+
+import javax.annotation.Nullable;
+
+/**
+ * Base Activity for React Native applications.
+ */
+public abstract class ReactFragmentActivity extends FragmentActivity
+        implements DefaultHardwareBackBtnHandler, PermissionAwareActivity {
+
+    private final ReactActivityDelegate mDelegate;
```

Comment:
I would rename it to `delegate` or `activityDelegate`

Reply:
This file is just a copy of the original `ReactActivity` that extends from `FragmentActivity` instead of `Activity`, so I'd keep it as is.